src/medCore/gui/toolboxes/medToolBox.cpp

```diff
@@ -308,3 +308,39 @@ void medToolBox::toXMLNode(QDomDocument* doc, QDomElement* currentNode)
 	currentNode->appendChild(elmt);
 }
 
+void medToolBox::handleDisplayError(int error)
+{
+    switch (error)
+    {
+    case medAbstractProcess::PIXEL_TYPE:
+        displayPixelTypeError();
+        break;
+    case medAbstractProcess::DIMENSION:
+        displayDataDimensionError();
+        break;
+    case medAbstractProcess::MESH_TYPE:
+        displayMeshTypeError();
+        break;
+    }
+}
+
+void medToolBox::displayPixelTypeError()
+{
+    QString error = "Pixel type not yet implemented";
+    qDebug() << name() + ": " + error;
+    medMessageController::instance()->showError(error,3000);
+}
+
+void medToolBox::displayDataDimensionError()
+{
+    QString error = "This toolbox is designed to be used with 3D volumes";
+    qDebug() << name() + ": " + error;
+    medMessageController::instance()->showError(error,3000);
+}
+
+void medToolBox::displayMeshTypeError()
```

Comment:
I can factorize the qDebug and medMessageController into a function. Concerning displayPixelTypeError() and the other fct, we want to be able to call them from a toolbox, even it's not run by medRunnableProcess. The other possibility would be to be able to call handleDisplayError(int error). I'm going to see that.